src/org/labkey/test/components/react/ToggleButton.java

```diff
@@ -31,6 +31,10 @@ public WebDriver getDriver()
 
     public ToggleButton set(boolean enabled)
     {
+        // if we can't determine the current state of the toggle, then throw without proceeding
+        if (!isOn() && !isOff())
+            throw new IllegalStateException("Unable to determine the current state of the toggle.");
```

Comment:
I would put this check in `isOn`. That way, the `waitFor` after toggling will ensure its in a valid state.
```
    public boolean isOn()
    {
        String buttonCls = getComponentElement().getAttribute("class");
        if (buttonCls.contains("toggle-on"))
            return true;
        else if (buttonCls.contains("toggle-off"))
            return false;
        else
            throw new IllegalStateException("Unable to determine the current state of the toggle.");
    }
```

Reply:
I like it! Updated.